### ISSUE TITLE
[RULE] AZR-000196 false-positive for usage of API version > 2019-04-01

### ISSUE THREAD

**Issue by tonybaloney** [Azure/PSRule.Rules.Azure#2414]
### Existing rule

Azure.Storage.SecureTransfer

### Suggested rule

This rule is raising issues when `supportsHttpsTrafficOnly` is not set to true, but this value has been true by default since version 2019-04-01.

I got the error from template analyser when running with a BICEP of v2022-05-01:

```bicep
resource storage 'Microsoft.Storage/storageAccounts@2022-05-01' = {
  name: name
  location: location
  tags: tags
  kind: kind
  sku: sku
  properties: {
    accessTier: accessTier
    allowBlobPublicAccess: allowBlobPublicAccess
    allowCrossTenantReplication: allowCrossTenantReplication
    allowSharedKeyAccess: allowSharedKeyAccess
    defaultToOAuthAuthentication: defaultToOAuthAuthentication
    dnsEndpointType: dnsEndpointType
    minimumTlsVersion: minimumTlsVersion
    networkAcls: {
      bypass: 'AzureServices'
      defaultAction: 'Allow'
    }
    publicNetworkAccess: publicNetworkAccess
  }
```

### Pillar

Security

**Comment by BernieWhite** [comment]
Thanks for flagging this @tonybaloney.
